### PR TITLE
Update build cmd to check tx inputs

### PR DESCRIPTION
**Pull request by Jimbo4350**
Solves #2996 and #3112

### CODE REVIEW

**Review by newhoggy — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs [hunk 1]
```diff
@@ -233,6 +236,13 @@ renderShelleyTxCmdError err =
     ShelleyTxCmdByronEraQuery -> "Query not available in Byron era"
     ShelleyTxCmdLocalStateQueryError err' -> renderLocalStateQueryError err'
     ShelleyTxCmdBalanceTxBody err' -> Text.pack $ displayError err'
+    ShelleyTxCmdExpectedKeyLockedTxIn txins ->
+      "Expected key witnessed tx inputs but got script witnessed tx inputs: " <>
+      Text.singleton '\n' <>
+      Text.intercalate (Text.singleton '\n') (map renderTxIn txins)
+    ShelleyTxCmdTxInsDoNotExist txins ->
+      "The following tx input(s) were not present in the UTxO: " <>
```
Comment (james-iohk) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:244]: Imo it's easier to read if each missing inout is on a new line (not just from the 2nd one), so add new line at end of message.

**Comment by james-iohk** [comment]
I've noticed that if the `tx-in-collateral` utxo is missing then only this is listed, even if the `tx-in` utxos are also missing. Then after fixing collateral input the same error is displayed with the missing `tx-in` utxos.  I kinda get why this would happen, is it possible or even worth making it list them all together?

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs:244]: I thought I fixed this. Good catch I'll re-push and bors.

**Pull request merged**